.github/workflows/test-and-lint.yml

```diff
@@ -0,0 +1,115 @@
+# .github/workflows/test-and-lint.yml
+
+name: Test and Lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # This job runs all the linting and formatting checks in parallel on a single OS.
+  lint:
+    name: Lint and Format Checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # --- Rust toolchain setup ---
+      - name: Set up Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt, clippy
+
+      # --- Minimum supported rust version --- 
+      - name: Set up MSRV checker
+        uses: taiki-e/install-action@cargo-hack
+
+      # --- Python toolchain setup for mypy ---
+      - name: Set up Python for linting
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11' # Use a specific version for linting
```

Comment:
3.11 is kind of an arbitrary version to use, isn't it? It isn't either the oldest we support neither the latest available.
Wouldn't it be better to setup the oldest Python version we support instead? The one we have specified in our `pyproject.toml` file.

Reply:
We could yeah. There are instances where we use `3.9` and `3.13`. This was just an arbitrary selection though.